src/material/slider/slider.ts

```diff
@@ -555,10 +556,22 @@ export class MatSlider
     transformOrigin: string;
   }): void {
     const trackStyle = this._trackActive.nativeElement.style;
+    const animationOriginChanged =
+      styles.left !== trackStyle.left && styles.right !== trackStyle.right;
+
     trackStyle.left = styles.left;
     trackStyle.right = styles.right;
-    trackStyle.transform = styles.transform;
     trackStyle.transformOrigin = styles.transformOrigin;
+
+    if (animationOriginChanged) {
```

Comment:
It's a little ugly but I got this working. The issue with the slider is that changes to properties automatically triggers UI updates. I added a boolean for checking whether the props has been initialized so that those UI updates don't get automatically triggered until the properties are done getting set up